### PR TITLE
Signing namespace

### LINKED ISSUE #18: Signing - validateSignatures

### LINKED ISSUE #140: Signing as a namespace

### PR DESCRIPTION
**Pull request by thibault-martinez**
Mainly close #140.

Will close #18, #19, #20, #21, #22 and #23 too.

### CODE REVIEW

File: external/keccak/source/KeccakP-1600-reference.cpp [hunk 1]
```diff
@@ -231,26 +229,6 @@ KeccakP1600_Permute_24rounds(void *state) {
...
-
-static void
-fromWordsToBytes(unsigned char *state, const tKeccakLane *stateAsWords) {
-  unsigned int i, j;
-
-  for (i = 0; i < nrLanes; i++)
-    for (j = 0; j < (64 / 8); j++)
-      state[i * (64 / 8) + j] = (unsigned char)((stateAsWords[i] >> (8 * j)) & 0xFF);
-}
-
```
Comment (thibault-martinez) [inline external/keccak/source/KeccakP-1600-reference.cpp]: Unused functions that generated warnings.

File: include/iota/crypto/signing.hpp [hunk 2]
```diff
@@ -28,34 +28,27 @@
...
-                                 const std::vector<Types::Trytes>& signatureFragments,
-                                 const Types::Trytes&              bundleHash);
-
-private:
-  // TODO Sponge
-  Kerl kerl_;
-};
+namespace Signing {
+
+Types::Trits key(const Types::Trytes& seed, const unsigned int& index = 0,
```
Comment (Cylix) [inline include/iota/crypto/signing.hpp]: maybe same stuff as previous PR, if you can include additional doc for these ones, would be perfect

Comment (thibault-martinez) [reply include/iota/crypto/signing.hpp]: There are specific issues for that, but sure, I'll close them all in this PR later today.

Comment (Cylix) [reply include/iota/crypto/signing.hpp]: ok no worry, I'll merge this one and just do the other specific issue when you want

**Pull request merged**